File: docs/src/main/scala/sexamples/basics/pingpongtimer/Parent.scala
package sexamples.basics.pingpongtimer

import se.sics.kompics.sl._
import se.sics.kompics.timer.Timer
import se.sics.kompics.timer.java.JavaTimer

class Parent extends ComponentDefinition {
  val pinger = create[Pinger];
  val pinger2 = create[Pinger];
  val ponger = create[Ponger];

  val timer = create[JavaTimer];

  connect(PingPongPort)(ponger -> pinger);
  connect(PingPongPort)(ponger -> pinger2);
  connect[Timer](timer -> pinger);
  connect[Timer](timer -> pinger2);

}

File: docs/src/main/java/jexamples/basics/pingpongtimer/Parent.java
package jexamples.basics.pingpongtimer;

import se.sics.kompics.Channel;
import se.sics.kompics.ComponentDefinition;
import se.sics.kompics.Component;
import se.sics.kompics.Init;
import se.sics.kompics.timer.Timer;
import se.sics.kompics.timer.java.JavaTimer;

public class Parent extends ComponentDefinition {
  Component pinger = create(Pinger.class, Init.NONE);
  Component ponger = create(Ponger.class, Init.NONE);
  Component pinger2 = create(Pinger.class, Init.NONE);
  Component timer = create(JavaTimer.class, Init.NONE);

  {
    connect(
        pinger.getNegative(PingPongPort.class),
        ponger.getPositive(PingPongPort.class),
        Channel.TWO_WAY);
    connect(pinger.getNegative(Timer.class), timer.getPositive(Timer.class), Channel.TWO_WAY);
    connect(
        pinger2.getNegative(PingPongPort.class),
        ponger.getPositive(PingPongPort.class),
        Channel.TWO_WAY);
    connect(pinger2.getNegative(Timer.class), timer.getPositive(Timer.class), Channel.TWO_WAY);
  }
}
